### PR TITLE
Integratord port to 4 4

### LINKED ISSUE #2944: Create tests for different types of inputs for the integrator module
Version | Component | Core issue | Core development branch | QA development branch 
-- | -- | -- | --|--|
4.3  | Integrator  | https://github.com/wazuh/wazuh/issues/12664   |  4.3 

# Description

As part of the fix introduced in https://github.com/wazuh/wazuh/issues/12664, we need to create some tests to validate it.

These are some scenarios that need to be tested:

- If the alert read is valid JSON, the alert is parsed and returned.

- If the alert read is invalid JSON, an error message is displayed and NULL is returned.

> "Invalid JSON alert read from 'logs/alerts/alerts.json': 'Invalid JSON alert example'"

- If the alert read is too long (more than 64Kb), an error message is displayed and NULL is returned.

> "Overlong JSON alert read from 'logs/alerts/alerts.json'"

- If the file cannot be read, and it is none of the above cases, check if the inode changed and try again. For this case, it will retry infinitely until it can read a complete line.

> "Can't read from 'logs/alerts/alerts.json'. Trying again"

Take in mind that the input of integrator module is the `alerts.json` file.

## Configuration option
- **Local Internal Options** : `syscheck.debug=2 integrador.debug=2`
- Configure folder to monitor with syscheck on realtime
- Configure virustotal integration
   ```xml
     <integration>
       <name>virustotal</name>
       <api_key>VIRUS_TOTAL_API_KEY</api_key> 
       <rule_id>100200,100201</rule_id>
       <alert_format>json</alert_format>
     </integration>
   ```
# Use cases

- [x] Case 1 - Insert valid alert:

    - Insert a valid alert into `alerts.json`
    - Wait for virustotal response in `alerts.json`

- [x] Case 2 - Insert invalid alert:

    - Insert a invalid alert into `alerts.json` (broken json)
    - Wait for error response in `ossec.log`
        - `"Invalid JSON alert read from 'logs/alerts/alerts.json': 'Invalid JSON alert example'"`

- [x] Case 3 - Insert Overlong alert (64kb+):

    - Insert an overlong alert into `alerts.json`
    - Wait for error response in `ossec.log`
        - `"Overlong JSON alert read from 'logs/alerts/alerts.json'"`

- [x] Case 4 - alerts.json cannot be read:

    - Generate continued alerts for integrator to keep reading the alerts.json
       - Insert content continually into file in the monitored folder
    - Create copy of `alerts.json`
    - Replace `alerts.json` with copy
    - Wait for error response in `ossec.log`
        - `"Can't read from 'logs/alerts/alerts.json'. Trying again"`
        -  `"Alert file inode changed. Reloading."`
                
# Development stage

- [x] In research
- [x] In development
    - [x] Linux 
- [x] In testing phase
    - [x] Linux 
- [x] In Review
- [ ] Completed

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| Closes #2944 |

## Description

This PR aims to add coverage to the Integratord module, by generating alerts and checking the responses on the ossec.log to 4.4.

### Added
The following test modules were added as part of the the new `test_integratord` test suite
- test_integratord_change_inode_alert.py
- test_integratord_read_json_alerts.py
- test_integratord_read_json_file_deleted.py

---

## Testing performed
| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09 (Developer)  |  test_integratord    | [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/30619/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/30621/) [:green_circle:](https://ci.wazuh.info/view/Tests/job/Test_integration/30622/)  | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/9300060/R1.zip) [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/9300061/R2.zip) [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/9300062/R3.zip) |  Centos  |  [ae2ab34](https://github.com/wazuh/wazuh-qa/pull/3163/commits/ae2ab342b038c5734d879efc81f9d6dcb8b14e51)  |  |
| Seyla (Reviewer)   |  test_integratord      | [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/30632/) [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/30633/) [🟢](https://ci.wazuh.info/view/Tests/job/Test_integration/30634/)  | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |  Centos  |  [ae2ab34](https://github.com/wazuh/wazuh-qa/pull/3163/commits/ae2ab342b038c5734d879efc81f9d6dcb8b14e51)  |  |

### CODE REVIEW

File: tests/integration/test_integratord/test_integratord_change_inode_alert.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+brief: Integratord manages wazuh integrations with other applications such as Yara or Virustotal, by feeding
+the integrated aplications with the alerts located in alerts.json file. This test module aims to validate that
+given a specific alert, the expected response is recieved, depending if it is a valid/invalid json alert, an
+overlong alert (64kb+) or what happens when it cannot read the file because it is missing.
+components:
+    - integratord
+suite: integratord_read_json_alerts
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integratord_change_inode_alert.py:13]:
```suggestion
suite: integratord_change_inode_alert
```

File: tests/integration/test_integratord/test_integratord_read_json_file_deleted.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+brief: Integratord manages wazuh integrations with other applications such as Yara or Virustotal, by feeding
+the integrated aplications with the alerts located in alerts.json file. This test module aims to validate that
+given a specific alert, the expected response is recieved, depending if it is a valid/invalid json alert, an
+overlong alert (64kb+) or what happens when it cannot read the file because it is missing.
+components:
+    - integratord
+suite: integratord_read_json_alerts
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integratord_read_json_file_deleted.py:13]:
```suggestion
suite: integratord_read_json_file_deleted
```

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_integratord/test_integratord_change_inode_alert.py (hunk 1 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integratord_change_inode_alert.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/3163/commits/c6cf2eaa2eff9021bf970eaa888578eb779d0177

File: tests/integration/test_integratord/test_integratord_read_json_file_deleted.py (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integratord_read_json_file_deleted.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/3163/commits/c6cf2eaa2eff9021bf970eaa888578eb779d0177

**Review by damarisg — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ Wazuh commit: TBD \
 Release report: TBD
 
 ### Added
-
+- Add Integratord IT - new test_integratord suite ([#3163](https://github.com/wazuh/wazuh-qa/pull/2957) - (Framework + Tests)
```
Comment (jmv74211) [inline CHANGELOG.md:11]: This was added in 4.3, right?

File: tests/integration/test_integratord/test_integratord_read_json_file_deleted.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    wazuh_monitor = FileMonitor(LOG_FILE_PATH)
+    command = f"touch {ALERT_FILE_PATH} && chmod 640 {ALERT_FILE_PATH} && chown wazuh:wazuh {ALERT_FILE_PATH}"
+
+    remove_file(ALERT_FILE_PATH)
+    check_integratord_event(file_monitor=wazuh_monitor, timeout=global_parameters.default_timeout*2,
+                            callback=generate_monitoring_callback(integrator.CB_CANNOT_RETRIEVE_JSON_FILE),
+                            error_message=integrator.ERR_MSG_CANNOT_RETRIEVE_MSG_NOT_FOUND)
+    # Create file and insert alert. Wait one second so Integrator detects the file before the insertion
+    run_local_command_returning_output(command)
+    time.sleep(2)
```
Comment (jmv74211) [inline tests/integration/test_integratord/test_integratord_read_json_file_deleted.py:122]: Always add in a comment the reason for each sleep

File: tests/integration/test_integratord/test_integratord_read_json_file_deleted.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    run_local_command_returning_output(f"echo '{metadata['alert_sample']}' >> {ALERT_FILE_PATH}")
+
+    # Read Response in ossec.log
+    check_integratord_event(file_monitor=wazuh_monitor, timeout=global_parameters.default_timeout*2,
+                            callback=generate_monitoring_callback(integrator.CB_VIRUSTOTAL_ALERT),
```
Comment (jmv74211) [inline tests/integration/test_integratord/test_integratord_read_json_file_deleted.py:127]: Add comment

File: tests/integration/test_integratord/test_integratord_read_json_file_deleted.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+def test_integratord_read_json_file_deleted(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                                            configure_local_internal_options_module, restart_wazuh_daemon_function,
+                                            wait_for_start_module):
+    '''
+    description: Check that if while integratord is reading from the alerts.json file, it is deleted, the expected
+    error message is displayed, and if the file is created again and alerts are inserted, integratord continues
+    working and alerts are read
+    wazuh_min_version: 4.3.5
+    tier: 1
+    parameters:
```
Comment (jmv74211) [inline tests/integration/test_integratord/test_integratord_read_json_file_deleted.py:81]: Add `test_phases` section

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_integratord/test_integratord_read_json_file_deleted.py (hunk 5 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integratord_read_json_file_deleted.py:153]: done in https://github.com/wazuh/wazuh-qa/pull/3163/commits/6b7c09d4157c76bd09758773696f5ececa5bf18a

File: CHANGELOG.md (hunk 3 above)
Comment (Deblintrake09) [reply CHANGELOG.md]: Yes. Removed from changelog requested

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**